src/Cli/PhpCommands.php

```diff
@@ -279,9 +184,23 @@ protected function versionExists($version)
         }
 
         // For network urls, run `curl -I` to do just a HEAD request.
-        // -s is "silent mode".
-        exec("curl -s -I $url", $output, $status);
-        return (strpos($output[0], '200 OK') !== false);
+        // -s is "silent mode", and -L follows redirects.
+        exec("curl -s -L -I $url", $output, $status);
```

Comment:
The only necessary change in this whole PR is to add the `-L` to `curl` here to prevent the false failure when the php ftp server returns a redirect. Everything else is opportunistic / engineering excellence work.